### DISCUSSION TITLE
Use new Response.json() static method

### DISCUSSION THREAD

**Discussion by nakleiderer** [remix-run/remix#3308] [Proposals]
There may be an opportunity to bring Remix closer to the web platform by utilizing the new `Response.json()` static method that was recently added to the fetch specification. Deno recently added support. This might be a good time to deprecate the `json` helper exported by Remix as well.

See:
- https://deno.com/blog/v1.22#new-responsejson-static-method
- https://github.com/whatwg/fetch/pull/1392/files 
- https://github.com/whatwg/fetch/issues/1389

There's might be a similar method for `Response.redirect()` as well, but it doesn't support passing in a `ResponseInit`: https://developer.mozilla.org/en-US/docs/Web/API/Response/redirect

```js
export let loader = () => {
  return Response.json({ hello: "world" }, { headers: { "x-something": "yup" } });
};

export let loader = () => {
  // Too bad it doesn't take a ResponseInit :(
  return Response.redirect("https://example.com", 307 );
};
```

**Comment by nakleiderer** [comment]
I just realized that @mjackson already commented on the whatwg issue: https://github.com/whatwg/fetch/issues/1389#issuecomment-1024838270

**Comment by sergiodxa** [comment]
This should be super simple to add, we could also support Response.redirect and still have redirect as a way to create one with the ResponseInit if you need it.

**Comment by mjackson** [comment]
You're certainly welcome to use `Response.json()` in Remix. It's more or less equivalent to what `json` already provides. I'm in no hurry to deprecate `json` though. Lots of people are already using it and deprecating it doesn't really do anything for them.

It's a shame `Response.redirect` doesn't take a `ResponseInit` as a 2nd parameter. It's useful for setting headers like `Set-Cookie`.

**Reply by sergiodxa** [reply]
The spec author of Response.redirect clearly didn't think about using it server-side.
